node-scripts/src/pipeline3/impl/dockerCompose.js

```diff
@@ -0,0 +1,249 @@
+      return "";
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  /**
+   * Util function that wraps the passed commands so each is applied either accordingly.
+   *
+   * @param {String} containerName - The name of the container.
+   * @param {String} ifExistsCommand - The command that should run if the container exists.
+   * @param {String} elseCommand - The command that will run if the container does *not* exist.
+   *
+   * @return {String} The script as a string.
+   */
+  ifExists: function() {},
+  restart: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker"
+      )
+      .toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose -p " + instanceDef.name + " restart ";
+
+    return script + "\n";
+  },
+  remove: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker"
+      )
+      .toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose -p " + instanceDef.name + " down -v ";
+
+    return script + "\n";
+  },
+  pull: function() {},
+  exec: (instanceDef, command, service) => {
+    let script = "";
+    script += "set -e\n";
+    var path = require("path");
+    let distPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker"
+      )
+      .toString();
+    script += "cd " + distPath + " && ";
+    script +=
+      "docker-compose -p " +
+      instanceDef.name +
+      " exec -T " +
+      service +
+      " /bin/bash -s <<" +
+      heredoc_2 +
+      "\n";
+    script += "set -e\n";
+    script += command + "\n";
+    script += heredoc_2;
+
+    return script + "\n";
+  },
+  setProperties: function(instanceDef, property, output) {
+    let script = "";
+    let path = require("path");
+    let propPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker/properties"
+      )
+      .toString();
+    let propFilePath = path.resolve(propPath, property.filename).toString();
+    script += require("../scripts").remote(
+      instanceDef.deployment.host.value,
+      "\n" +
+        "if [[ ! -e " +
+        propFilePath +
+        " ]]; then\n" +
+        "sudo mkdir -p " +
+        propPath +
+        "\n" +
+        "sudo touch " +
+        propFilePath +
+        "\n" +
+        "fi\n" +
+        "sudo bash -c 'cat > " +
+        propFilePath +
+        " <<EOF \n" +
+        output +
+        "\nEOF'\n"
+    );
+    return script;
+  },
+  setLinks: function() {},
```

Comment:
Seems like this is not tested right? This should probably be unit tested in **dockerCompose.spec.js**